typescript/web/src/components/labeling-tool/options-tool-bar/edit-selection-mode/toggle-button-group.tsx

```diff
@@ -0,0 +1,50 @@
+import { ButtonGroup, ButtonGroupProps, useRadioGroup } from "@chakra-ui/react";
+import * as React from "react";
+import { ToggleButtonProps } from "./toggle-button";
+
+interface ToggleButtonGroupProps<T> extends Omit<ButtonGroupProps, "onChange"> {
+  name?: string;
+  value: T;
+  defaultValue?: string;
+  onChange?: (value: T) => void;
+}
+
+export const ToggleButtonGroup = <T extends string>(
+  props: ToggleButtonGroupProps<T>
+) => {
+  const { children, name, defaultValue, value, onChange, isDisabled, ...rest } =
+    props;
+  const { getRootProps, getRadioProps } = useRadioGroup({
+    name,
+    defaultValue,
+    value,
+    onChange,
+  });
+
+  const buttons = React.useMemo(
+    () =>
+      React.Children.toArray(children)
+        .filter<React.ReactElement<ToggleButtonProps>>(React.isValidElement)
+        .map((button, index, array) => {
+          const isFirstItem = index === 0;
+          const isLastItem = array.length === index + 1;
+
+          const styleProps = {
+            ...(isFirstItem && !isLastItem ? { borderRightRadius: 0 } : {}),
+            ...(!isFirstItem && isLastItem ? { borderLeftRadius: 0 } : {}),
+            ...(!isFirstItem && !isLastItem ? { borderRadius: 0 } : {}),
+            ...(!isLastItem ? { mr: "-px" } : {}),
+          };
+
+          return React.cloneElement(button, {
```

Comment:
This is not really the most "react-ish" way to do this but I guess it works.

Usually, we tend to avoid this syntax because it may get odd when you debug the children. Somewhere they are written like `<child propA propB />` and somehow, when you log their inputs they have an additional `propC`. Which comes from a clone element in the parent. 

One common alternative I would see is to call the buttons directly here:
```tsx
 <ButtonGroup>
   <Button first />
   <Button />
   <Button last />
 </ButtonGroup>
```

But it also implies that you are here giving all the necessary props to the buttons. So, you need to make sure `ToggleButtonGroup` is given all the necessary props to forward them.